project/npda/views/submissions.py

```diff
@@ -163,16 +163,19 @@ def post(self, request, *args, **kwargs):
                 new_first.save()
             messages.success(request, "Cohort submission deleted successfully")
 
-        if button_name == "download-data":
+        submission = None
+
+        if request.POST.get("audit_id"):
             submission = Submission.objects.filter(
                 pk=request.POST.get("audit_id")
             ).get()
+        else:
+            submission = self.get_queryset().order_by("-submission_date").first()
```

Comment:
Should again use the manager method?

Reply:
We have the submission in the context now so it can just use the hidden input field like the other